api/turing/cluster/servicebuilder/service_builder.go

```diff
@@ -434,3 +460,23 @@ func buildLabels(
 	}
 	return labeller.BuildLabels(r)
 }
+
+// mergeEnvVars merges multiple sets of environment variables and return the merging result.
+// All the EnvVars passed as arguments will be not mutated.
+// EnvVars to the right have higher precedence.
+func mergeEnvVars(left []corev1.EnvVar, rightEnvVars ...[]corev1.EnvVar) []corev1.EnvVar {
```

Comment:
Added this helper function that came from https://github.com/caraml-dev/merlin in order to merge env vars with replacement (for those with the same name) instead of appending blindly (which can create duplicated env vars in the list).